### ISSUE TITLE
make inbound response match Lendio's

### ISSUE THREAD

**Issue by cgrayson** [activeprospect/leadconduit-integration-lendio#63]
We missed this in the previous changes, but I think (correct me if I'm wrong, @alexj) that the response provided to vendors from the LC Lendio integration needs to match what's provided by Lendio's own API. See the section on "Sample Success Response" in [this document](https://drive.google.com/a/activeprospect.com/file/d/0B3snTzjBrjGRNTNBUThNaUFxNkE/view) for details.

I'm not sure if the entire thing, especially echoing back all the lead data, is required. But it seems that fields like `error`, `code`, and `redirect_url` should definitely be in that format.

**Comment by zalary** [comment]
@cgrayson I saw that but didn't fully grep it - but yes, that makes sense.  I'll work on getting this updated.  

@alexj - I'm not sure who else is using this integration (maybe just PFS?) but maybe it's worth reaching out to anyone using it and make sure the change in format of response will be okay for them as well.

**Comment by cgrayson** [comment]
> @alexj - I'm not sure who else is using this integration (maybe just PFS?) but maybe it's worth reaching out to anyone using it and make sure the change in format of response will be okay for them as well.

I don't think anyone is actively using this inbound integration yet, so that should be okay. 

Zal & I discussed it, and we think the best way forward is to make the outbound integration append the entire JSON response from Lendio, which the inbound's `response()` function could then simply return as-is. 

That function would also need to account for the possibility that no Lendio response JSON is available, as in the case of Lendio being down or a lead being rejected prior to submission to Lendio. I think mimicking the "Sample Error Response" in that posting doc would be a good place to start for the format for that case: 

```json
{
"error":true,
"code":422,
"payment":"0.00",
"message":"Borrower application rejected. time_in_business is required" 
}
```

**Comment by alexj** [comment]
Correct, no one is using it yet. PFS will be the first. Your solution sounds good to me since it matches their posting doc. Thanks for catching this!

**Comment by alexkwolfe** [comment]
My vote would be for appending the structured data, rather than appending the response body as a string.

**Comment by cgrayson** [comment]
Okay, that sounds like a good approach, and not too difficult (I think we can use that appended structure as-is pretty easily). I'm thinking the variables will be declared in `response.variables` as usual, even though they aren't expected to be used explicitly by later steps. 

But a couple of the things are coming back as arrays, like `addresses` in this snippet: 

```json
"registration_date":"2015-09-29 17:31:53",
"business_name":"test",
"addresses":[
  {
    "street":"123 North Water Str",
    "street2":null,
    "city":"Corpus Christi",
    "county":null,
    "state_abbreviation":"UT",
    "postal_code":"84095",
    "type":"business"
  }
 ]
```

I'm not sure how that could be listed as a response variable, other than just: `{ name: 'addresses', type: 'array' … }`. Wonder if that will work.

**Comment by alexkwolfe** [comment]
I wonder too. :)
